room/room-compiler/src/main/kotlin/androidx/room/processor/ProcessorErrors.kt

```diff
@@ -175,9 +175,9 @@ object ProcessorErrors {
     val CANNOT_FIND_UPDATE_RESULT_ADAPTER = "Not sure how to handle update method's " +
         "return type. Currently the supported return types are void, int or Int."
 
-    val USING_MULTIPLE_CONCURRENCY_PATTERNS = "Dao methods that have a suspend modifier " +
-        "shouldn't return another async type. Most probably this is an error. Consider " +
-        "changing the return type or removing the suspend modifier."
+    fun suspendReturnsDeferredType(returnTypeName: String) = "Dao functions that have a suspend " +
+        "modifier must not return deferred/async return type $returnTypeName. Most probably this " +
```

Comment:
nit: extra 'return' here

Maybe change to `must not contain a deferred/async return type`
or `must not return a deferred/async type`

and then move the input name to a parenthesis `($returnTypeName)`

Reply:
Done, thank you for the review Dany!